pyflakes/test/test_imports.py

```diff
@@ -564,6 +563,15 @@ def bar():
         fu
         ''', m.UnusedImport, m.UndefinedName)
 
+    def test_shadowedInNestedScope(self):
+        self.flakes('''
+        import fu
+        def bar():
+            import fu
+            fu
+        fu
+        ''', m.ImportShadowedByImport)
```

Comment:
Differentiating between types of imports will be much cleaner and simpler after #54